dev_guide/builds/index.adoc

```diff
@@ -0,0 +1,129 @@
+[[dev-guide-how-builds-work]]
+= How Builds Work
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[dev-guide-what-is-a-build]]
+== What Is a Build?
+
+A
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#builds[_build_]
+in {product-title} is the process of transforming input parameters into a
+resulting object. Most often, builds are used to transform source code into a
+runnable container image.
+
+A _build configuration_, or `BuildConfig`, is characterized by a _build strategy_ and one
+or more sources. The strategy determines the aforementioned process, while the
+sources provide its input.
+
+There are four build strategies:
+
+- Source-To-Image (S2I)
+(xref:../../architecture/core_concepts/builds_and_image_streams.adoc#source-build[description],
+xref:source-to-image-strategy-options[options])
+- Docker
+(xref:../../architecture/core_concepts/builds_and_image_streams.adoc#docker-build[description],
+xref:docker-strategy-options[options])
+- Pipeline
+(xref:../../architecture/core_concepts/builds_and_image_streams.adoc#pipeline-build[description],
+xref:pipeline-strategy-options[options])
+- Custom
+(xref:../../architecture/core_concepts/builds_and_image_streams.adoc#custom-build[description],
+xref:custom-strategy-options[options])
+
+And there are four types of build sources that can be given as input:
+
+- xref:build_inputs_output.adoc#source-code[Git]
+- xref:build_inputs_output.adoc#dockerfile-source[Dockerfile]
+- xref:build_inputs_output.adoc#image-source[Image]
+- xref:build_inputs_output.adoc#binary-source[Binary]
+
+It is up to each build strategy to consider or ignore a certain type of source,
+as well as to determine how it is to be used.
+
+Binary and Git are mutually exclusive source types. Dockerfile and Image can be
+used by themselves, with each other, or together with either Git or Binary.
+Also, the Binary build source type is unique from the other options in
+xref:build_inputs_output.adoc#binary-source[how it is specified to the system].
+
+[[defining-a-buildconfig]]
+== What Is a BuildConfig?
+
+A build configuration describes a single build definition and a set of
+xref:triggering_builds.adoc#build-triggers[triggers] for when a new build should
+be created. Build configurations are defined by a `BuildConfig`, which is a REST
+object that can be used in a POST to the API server to create a new instance.
+
+Depending on how you choose to create your application using {product-title}, a
+`BuildConfig` is typically generated automatically for you if you use the web
+console or CLI, and it can be edited at any time. Understanding the parts that
+make up a `BuildConfig` and their available options can help if you choose to
+manually tweak your configuration later.
```

Comment:
@bparees I added this 2nd paragraph here and renamed the section from "Defining a BuildConfig" -> "What Is a BuildConfig?" because it initially seemed potentially ike we were expecting you to go craft a BuildConfig from scratch before doing anything w/ builds. Hopefully this makes it a little clearer what the expectations are. Any suggestions on this?